Firestore/core/src/firebase/firestore/local/memory_mutation_queue.mm

```diff
@@ -21,7 +21,6 @@
 #import "Firestore/Protos/objc/firestore/local/Mutation.pbobjc.h"
```

Comment:
Theoretically, this class could avoid a lot of `MutationBatch` copies when returning batches satisfying certain criteria by returning pointers instead (or vectors of pointers). On the other hand, it likely outperforms the LevelDB queue (which has to return by value) by such a large margin that this optimization isn't worth it.

Reply:
I'm not sure this would be much of a win:
  * The mutations in the vectors already are copy optimized
  * Most mutation batches have only one mutation
  * Adding an internal shared_ptr would be pretty much the only recourse given the constraints of the leveldb interface, but this would likely cause more allocations than it saves given the above.